Code/src/main/java/nl/utwente/group10/haskell/type/Type.java

```diff
@@ -2,45 +2,29 @@
 
 import nl.utwente.group10.haskell.HaskellObject;
 
+import java.util.logging.Logger;
+
 /**
  * Abstract class for Haskell types. Provides an interface for common methods.
  *
  * Types should be immutable. Also, a Type subclass instance that represents the same in Haskell as another instance of
  * the same class should be equal, that is {@code this.equals(that)} should return {@code true}.
  */
-public abstract class Type extends HaskellObject implements Comparable<Type> {
+public abstract class Type extends HaskellObject {
+    /** Logger instance for types. */
+    protected Logger logger = Logger.getLogger(Type.class.getName());
+
     /**
-     * Tests whether the given type is usable within the context of this type.
-     *
-     * Examples:
-     * - The `Integer` type is valid within the type class `Num`
-     * - Every type is valid within a {@code VarT} type.
-     *
-     * @param other The type to test.
-     * @return Whether the given type is usable in place of this type.
+     * Prunes the type subtree and returns the resolved type for this type. The resolved type can be this type itself.
```

Comment:
This JavaDoc could use a nice example of what prune does, exactly. I'm half-sure I forgot it already. (Perhaps also call it instanced or maybe even inferred, not resolved.)